straxen/plugins/bayes_peak_classification.py

```diff
@@ -121,30 +130,63 @@ def compute_inference(bins: int, bayes_n_nodes: int, cpt: np.ndarray, n_bayes_cl
     waveform_num_bin_edges = len(waveform_bin_edges)
     quantile_bin_edges = bins[1, :][bins[1, :] > -1]
     quantile_num_bin_edges = len(quantile_bin_edges)
-
     waveform_values = np.digitize(waveforms, bins=waveform_bin_edges)-1
-    waveform_values[waveform_values < 0] = int(0)
-    waveform_values[waveform_values > int(waveform_num_bin_edges - 2)] = int(waveform_num_bin_edges - 2)
+    _set_2d_to_zero_or_max_val(waveform_values,
+                               min_val=0,
+                               max_val=np.int64(waveform_num_bin_edges - 2))
 
     quantile_values = np.digitize(quantiles, bins=quantile_bin_edges)-1
-    quantile_values[quantile_values < 0] = int(0)
-    quantile_values[quantile_values > int(quantile_num_bin_edges - 2)] = int(quantile_num_bin_edges - 2)
-
-    values_for_inference = np.append(waveform_values, quantile_values, axis=1)
-
-    # Inference
-    distributions = [[] for i in range(bayes_n_nodes*2)]
-    for i in np.arange(0, bayes_n_nodes, 1):
-        distributions[i] = np.asarray(cpt[i, :waveform_num_bin_edges-1, :])
-    for i in np.arange(bayes_n_nodes, bayes_n_nodes * 2, 1):
-        distributions[i] = np.asarray(cpt[i, :quantile_num_bin_edges-1, :])
+    _set_2d_to_zero_or_max_val(quantile_values,
+                               min_val=0,
+                               max_val=np.int64(quantile_num_bin_edges - 2))
+
+    wf_posterior = _get_log_posterior(
+        nodes=bayes_n_nodes,
+        n_bins=waveform_num_bin_edges-1,
+        n_classes=n_bayes_classes,
+        cpt=cpt[:bayes_n_nodes],
+        wf_len=len(waveforms),
+        wf_values=waveform_values
+    )
+    quantile_posterior = _get_log_posterior(
+        nodes=bayes_n_nodes,
+        n_bins=quantile_num_bin_edges-1,
+        n_classes=n_bayes_classes,
+        cpt=cpt[bayes_n_nodes:],
+        wf_len=len(waveforms),
+        wf_values=quantile_values
+    )
 
     lnposterior = np.zeros((len(waveforms), bayes_n_nodes*2, n_bayes_classes))
-    for i in range(bayes_n_nodes*2):
-        lnposterior[:, i, :] = np.log(distributions[i][values_for_inference[:, i], :])
-
+    lnposterior[:, :bayes_n_nodes] = wf_posterior
+    lnposterior[:, bayes_n_nodes:] = quantile_posterior
```

Comment:
I think it make sense to group this together into a single `get_log_posterior` function to enhance readability.  And then we can add some proper documentation what exactly we are computing here. I think I made this comment already in the original PR @Aaron. Based on the code alone it is pretty hard to understand at least for me.

Reply:
Sure, I think this is a good idea. I must admit that I also don't fully understand from the code alone